### PR TITLE
feat(schematics): implement 'flat' flag for feature

### PR DESCRIPTION
**Pull request by theorlovsky**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Feature schematic doesn't have `flat` flag.

Issue Number: #393 

## What is the new behavior?

Feature schematic has `flat` flag.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by theorlovsky** [comment]
@NetanelBasal the build is failing, but it's not related to the changes in this PR.

File: docs/docs/angular/schematics.mdx [hunk 1]
```diff
@@ -48,6 +48,26 @@ ng g @datorama/akita:feature auth/auth --plain
 The akita-schematics: prefix is only needed when the default collection isn't set to akita-schematics
 :::
 
+### Generate a folder
+
+Add the `--flat false` to generate a feature inside its own folder. For example:
```
Comment (NetanelBasal) [inline docs/docs/angular/schematics.mdx:53]: It'll be better to add a concrete example as we have in the issue.

File: libs/akita-schematics/src/ng-g/utils/parse-name.ts [hunk 2]
```diff
@@ -6,13 +6,13 @@ export interface Location {
 }
 
 export function parseName(options: any): Location {
-  const { name, path, dirName, feature } = options;
+  const { name, path, dirName, feature, flat } = options;
   const nameWithoutPath = basename(name as Path);
   const namePath = dirname((path + '/' + name) as Path);
-  const normalizedPath = feature ? normalize('/' + namePath + '/' + dirName) : normalize('/' + namePath);
+  const normalizedPath = feature ? normalize('/' + namePath + '/' + dirName + (flat ? '' : '/' + nameWithoutPath)) : normalize('/' + namePath);
```
Comment (NetanelBasal) [inline libs/akita-schematics/src/ng-g/utils/parse-name.ts:12]: Can you refactor it please to make it more readable? Maybe separate it into variables and use template strings.

File: docs/docs/angular/schematics.mdx (hunk 1 above)
Comment (theorlovsky) [reply docs/docs/angular/schematics.mdx:53]: agree

File: libs/akita-schematics/src/ng-g/utils/parse-name.ts (hunk 2 above)
Comment (theorlovsky) [reply libs/akita-schematics/src/ng-g/utils/parse-name.ts:12]: sure, will do it tomorrow

File: docs/docs/angular/schematics.mdx (hunk 1 above)
Comment (theorlovsky) [reply docs/docs/angular/schematics.mdx]: done

File: libs/akita-schematics/src/ng-g/utils/parse-name.ts (hunk 2 above)
Comment (theorlovsky) [reply libs/akita-schematics/src/ng-g/utils/parse-name.ts]: done

**Pull request merged**